mobile/lib/modules/home/services/asset_cache.service.dart

```diff
@@ -8,17 +8,24 @@ class AssetCacheService extends JsonCache<List<Asset>> {
   AssetCacheService() : super("asset_cache");
 
   @override
-  void put(List<Asset> data) {
-    putRawData(data.map((e) => e.toJson()).toList());
+  void put(List<Asset> data) async {
+    serialize(List<Asset> assets) {
```

Comment:
Should we put the `serialize` method as a method of the class instead of the function? I believe we haven't used this nested function declaration on the mobile code base anywhere

Reply:
A `static` class method should work well for `serialize`

Reply:
that also would catch the error that the method currently uses `data` instead of its parameter `assets`